crates/llm-base/src/inference_session.rs

```diff
@@ -946,3 +928,24 @@ pub fn feed_prompt_callback<'a, E: std::error::Error + 'static>(
         None => Ok(InferenceFeedback::Continue),
     }
 }
+
+/// Create the memory K/V tensors for the inference-session.
+fn kv_memory(
+    context: &Context,
+    config: &InferenceSessionConfig,
+    n_elements: usize,
+) -> (Tensor, Tensor) {
+    let memory_k = context
+        .new_tensor_1d(config.memory_k_type.into(), n_elements)
+        .set_name("memory_k");
+    let memory_v = context
+        .new_tensor_1d(config.memory_v_type.into(), n_elements)
+        .set_name("memory_v");
+
+    if config.use_gpu {
+        ggml::accelerator_offload_tensor_no_scratch(&memory_k);
+        ggml::accelerator_offload_tensor_no_scratch(&memory_v);
```

Comment:
Should this be part of the `Tensor` interface too?

Reply:
This is very specific for CUDA and the K/V-memory. No other Tensors should be offloaded without scratch.

Reply:
In that case, add a comment to that effect so that we remember the reasoning for this.